hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -131,7 +133,9 @@ public void run(Runnable task) {
     }
 
     private boolean publishCurrentTask() {
-        return (getPartitionId() != AD_HOC_PARTITION_ID && currentTask == null);
+        boolean isClientRunnable = currentTask instanceof MessageTask
```

Comment:
the problem with this approach is that when we introduce a new client runnable which gets executed by operation-runner then we will silently introduce this very same bug. We can say that with the new protocol things will not change that much but I think it is still a possibility. We need a general interface here for operation/client-message but that's not the subject of this PR